Tensile/SolutionStructs.py

```diff
@@ -3285,9 +3287,19 @@ def assignDerivedParameters(state):
     #if state["KernelLanguage"] == "Assembly" and state["PackSummationDims"]:
     #    reject(state, "PackSummationDims does not yet support assembly")
 
-    # Default GlobalReadVectorWidth
-    if state["GlobalReadVectorWidth"] == -1:
-      state["GlobalReadVectorWidth"] = state["VectorWidth"]
+    # Use GlobalReadVectorWidth if it is not -1
+    if state["GlobalReadVectorWidth"] != -1:
+      state["GlobalLoadVectorWidthA"] = state["GlobalReadVectorWidth"]
+      state["GlobalLoadVectorWidthB"] = state["GlobalReadVectorWidth"]
+
+    # Default GlobalLoadVectorWidthA, B
+    for tc in ('A','B'):
+      if state["GlobalLoadVectorWidth%s"%tc] == -1:
+        state["GlobalLoadVectorWidth%s"%tc] = state["VectorWidth"]
+      # reject - GLVWA/B too big
+      if (state["GlobalLoadVectorWidth%s"%tc] * state["ProblemType"]["DataType"].numBytes()) > 16:
+        reject(state, "GlobalLoadVectorWidth%s * DataType.numBytes() > 16"%tc)
+        return
```

Comment:
What if user sets both GlobalRead and GlobalLoad? Should there be a reject case for this to avoid duplicate kernels?

Reply:
We just prioritize GlobalReadVectorWidth in this case.
This is to prioritize existing kernels with GlobalReadVectorWidth > 0.
I already added some explanation in Common.py about this logic.

Reply:
We do not generate duplicated kernels in that caes.
It is because GlobalReadVectorWidth is assigned to both GlobalLoadVectorWidthA and B.
Only 1 kernel per same GlobalReadVectorWidth (regardless of GlobalLoadVectorWidthA, B values).

Reply:
In rocblas logic files, all GlobalReadVectorWidth , GlobalLoadVectorWidthA, GlobalLoadVectorWidthB will be valid values (>0) if we use exisiting logic (with GlobalReadVectorWidth ).
All existing kernels in rocblas yaml will be rejected if we add such reject condition here.

Reply:
rocBLAS logic files shouldn't get rejected because liblogic doesn't go through SolutionStructs the same way that tuning files do. Anything that makes it to liblogic gets built. I'm just wondering if there's anything preventing duplicate kernels from getting generated if someone sets all 3 parameters in a tuning file? I see that GRWV overrides values GLVWa/b, but are GLVWa/b included in kernel name?

Reply:
I encountered some kernel generation error due to being rejected.
So, I am assuming TensileCreateLibrary goes through SolutionStruct code.

No duplicated kernel will be generated even if we keep both parameters (GRVW and GLVWA,B) without rejecting them.
Duplicated kernels are generated only if they have different kernel names with same parameters.
Since GLVWA,B are overwritten with GRVW, Tensile generate only 1 kernel per GRVW regardless of the values we set for GLVWA,B in Tensile yaml.

GLVWA,B will be in kernel name after my change.

Reply:
Thanks for the clarification